packages/netlify-cms-core/src/reducers/editorialWorkflow.ts

```diff
@@ -121,13 +140,48 @@ const unpublishedEntries = (state = Map(), action: EditorialWorkflowAction) => {
       );
 
     case UNPUBLISHED_ENTRY_PUBLISH_SUCCESS:
+      return state.withMutations(map => {
+        map.deleteIn(['entities', `${action.payload!.collection}.${action.payload!.slug}`]);
+        map.deleteIn([
+          'entities',
+          `${action.payload!.collection}.${action.payload!.slug}`,
+          'isPublishing',
+        ]);
+      });
+
     case UNPUBLISHED_ENTRY_PUBLISH_FAILURE:
       return state.withMutations(map => {
         map.deleteIn(['entities', `${action.payload!.collection}.${action.payload!.slug}`]);
+        map.setIn(
```

Comment:
Not sure why this is required. It seems to conflict with the previous line of deleting the parent path.